### PR TITLE
Apply design changes to email login screen

### PR DESCRIPTION
**Pull request by steverydz**
## Done

Updated the email login form to the latest design

## QA
- Make sure you have all dependencies:
  - `snap install go`
  - `sudo apt install build-essential bzr git`
- Run the site `make server`
- Go to http://localhost:8081/login-email
- Try logging in with the form and check that it uses the [Vanilla validation pattern](https://vanillaframework.io/docs/base/forms#validation)

## Issue
Fixes https://github.com/canonical-web-and-design/snap-squad/issues/1838

## Screenshots

login_with_email

login_with_email_error

### CODE REVIEW

File: templates/authentication-required [hunk 1]
```diff
@@ -26,10 +26,10 @@
       <div class="col-6 col-start-large-4">
         <div class="p-card--highlighted">
           <div class="p-card__thumbnail">
-            <h1 class="p-heading--four">Login with</h1>
+            <h1 class="p-heading--four">Login {{ if .UseEmail }}with email address{{ end }}</h1>
           </div>
           <hr class="u-sv1">
-  {{if .Error}}
+  {{if and .Error  (not .UseEmail)}}
```
Comment (tbille) [inline templates/authentication-required]: remove extra space

```suggestion
  {{if and .Error (not .UseEmail)}}
```

File: templates/authentication-required [hunk 2]
```diff
@@ -38,20 +38,31 @@
...
-            <form method="post" action="{{ .WithEmailURL }}">
-              <label for="email">Email address</label>
-              <input type="email" id="email" name="email">
-              <button type="submit">Continue</button>
+            <form method="post" action="{{ .WithEmailURL }}" style="margin-bottom: 1rem;">
+              <div class="p-form-validation {{ if .Error  }}is-error{{ end }}">
+                <label for="email">Email address</label>
+                <div class="p-form--inline" style="width: 100%;">
+                  <div style="width: 100%;">
+                    <input type="email" id="email" name="email" class="p-form-validation__input">
```
Comment (tbille) [inline templates/authentication-required]: add `required` to prevent posting an empty string.

**Review by mhilton — APPROVED** [review]
LGTM (not that I'm an expert)

File: templates/authentication-required (hunk 2 above)
Comment (steverydz) [reply templates/authentication-required]: done

**Review by tbille — APPROVED** [review]
LGTM. Let's have a design review and should be good to go.

**Comment by spencerbygraves** [comment]
Thanks @steverydz, a few points below...

- In the wireframes Daniel had used "Log in", rather than "Login"
- The card is quite a bit wider than [the design](https://zpl.io/2jgxERQ)
- Are we not using logos anymore?

**Comment by steverydz** [comment]
@spencerbygraves 

I've narrowed it to four grid cols rather than six and updated the text. This PR is just for the email form. The SSO buttons are coming in another PR.

Screenshot from 2021-02-23 11-22-42

**Comment by spencerbygraves** [comment]
The width is still slightly different, we'll have to see if it works when we add the 3 logos across. Thanks.

**Pull request merged**